internal/feature/flags.go

```diff
@@ -0,0 +1,59 @@
+package feature
+
+import (
+	"fmt"
+	"sync"
+)
+
+// All feature flags need to be defined here with their default values
+var flags = map[string]bool{
+	"events": false,
+}
+
+type Checker interface {
+	IsEnabled(flag string) bool
+}
+
+type Writer interface {
+	Enable(flag string) error
+	Disable(flag string) error
+}
+
+func NewStaticMapFeature() *staticMapFeature {
+	return &staticMapFeature{flags: flags, mu: &sync.Mutex{}}
```

Comment:
it seems bad for all instances of this class to share a single global map? was that intentional? seems like we should dependency-inject this

Reply:
It was intentional: there are only two different kinds of feature flag sets we will ever want. One for testing and one for everything else. As a result it didn't seem necessary to dependency inject the flag set because we want to discourage people from providing their own.

I don't feel super strongly though and I'm happy to add it.

Reply:
talked about this offline. his will cause problems, because one test will enable the flag, and then the global state will affect other tests. let's copy the map so that we don't rely on global state